syntax = "proto3";

package message;

message RequestVoteReq{
	int32 term = 1;
	int32 candidateId = 2;
  int32 lastLogIndex = 3;
  int32 lastLogTerm = 4;
}

message RequestVoteRsp {
	int32 term = 1;
	bool voteGranted = 2;
}

message Entry{
	int32 term = 1;
	string command = 2;
}

message AppendEntriesReq {
	int32 term = 1;
	int32 prevLogIndex = 2;
	int32 prevLogTerm = 3;
  repeated Entry entries = 4;
  int32 leaderCommit = 5;
}

message AppendEntriesRsp {
	int32 term = 1;
	bool  success = 2;
  int32 expectIndex = 3;
  int32 expectTerm = 4;
}

service RaftService {
	rpc RequestVote(RequestVoteReq) returns (RequestVoteRsp) {}
	rpc AppendEntries(AppendEntriesReq) returns (AppendEntriesRsp) {}
}